lib/Doctrine/ODM/PHPCR/Query/Query.php

```diff
@@ -180,7 +188,7 @@ public function execute($parameters = null, $hydrationMode = null)
                 $data = $this->query->execute();
                 break;
             case self::HYDRATE_DOCUMENT:
-                $data = $this->dm->getDocumentsByPhpcrQuery($this->query, $this->documentClass, $this->primaryAlias);
+                $data = $this->dm->getDocumentsByPhpcrQuery($this->query, $this->documentClass);
```

Comment:
ooops. DocumentManagerInterface only has 2 parameters, but the actual method has 3. also the dm decorator does not have that 3rd parameter, so it will drop it when used.

we should probably fix that interface then, even though it means potentially breaking custom DMs . but dropping parameters is worse than such a break imo